src/seeds.bioregion.cpp

```diff
@@ -160,15 +165,33 @@ ACTION bioregion::join(name bioregion, name account) {
     auto mitr = members.find(account.value);
 
     if (mitr != members.end()) {
-        check(false, "user exists TODO allow users to switch");
-    } else {
-        members.emplace(_self, [&](auto & item) {
-            item.bioregion = bioregion;
+        if (mitr -> bioregion == bioregion) { return; }
+        leave(mitr -> bioregion, account);
+    }
+
+    uint64_t now = eosio::current_time_point().sec_since_epoch();
+
+    auto ditr = biodelays.find(account.value);
+    if (ditr == biodelays.end()) {
+        biodelays.emplace(_self, [&](auto & item){
             item.account = account;
+            item.apply_vote_delay = false;
+            item.joined_date_timestamp = now;
+        });
+    } else {
+        check(ditr -> joined_date_timestamp < now - (utils::moon_cycle * config_float_get("bio.vote.del"_n)), "user needs to wait until the delay ends");
```

Comment:
Since only 1 float config is being used, would it be just more clear to add a variable to store the config, instead of a function?

Reply:
@juliolrmonteiro config_get is our standard, as it takes care of the case where the setting was not yet set.